src/Features/Core/Portable/Shared/TestHooks/TaskExtensions.cs

```diff
@@ -21,7 +21,7 @@ public static Task CompletesAsyncOperation(this Task task, IAsyncToken asyncToke
 
         public static Task CompletesTrackingOperation(this Task task, IDisposable token)
         {
-            if (token == null)
+            if (token == null || token == EmptyAsyncToken.Instance)
```

Comment:
Why do we allow both null and EmptyAsyncToken.Instance? What's the intended difference?

Reply:
I have no idea. While I analyzed the `EmptyAsyncToken.Instance` case carefully, I didn't review the other callers of this method in detail, since it didn't show up as problematic during investigation.